src/NuGet.Core/NuGet.Protocol/Utility/OfflineFeedUtility.cs

```diff
@@ -193,12 +194,17 @@ public static void ThrowIfInvalid(string path)
                             ? PackageSaveMode.Defaultv3
                             : PackageSaveMode.Nuspec | PackageSaveMode.Nupkg;
 
-                        var versionFolderPathContext = new VersionFolderPathContext(
+                        var signedPackageVerifier = new SignedPackageVerifier(
+                            SignatureVerificationProviderFactory.GetSignatureVerificationProviders(),
+                            SignedPackageVerifierSettings.Default);
+
+                        var packageExtractionV3Context = new PackageExtractionV3Context(
                             packageIdentity,
                             source,
                             logger,
                             packageSaveMode: packageSaveMode,
-                            xmlDocFileSaveMode: PackageExtractionBehavior.XmlDocFileSaveMode);
+                            xmlDocFileSaveMode: PackageExtractionBehavior.XmlDocFileSaveMode,
+                            signedPackageVerifier: signedPackageVerifier);
```

Comment:
Can simplify the above 3 lines to:

    packageSaveMode, 
    PackageExtractionBehavior.XmlDocFileSaveMode, 
    signedPackageVerifier);